python/common/org/python/types/List.java

```diff
@@ -167,8 +168,14 @@ public List(org.python.Object [] args, java.util.Map<java.lang.String, org.pytho
             int count = Math.min(size, otherSize);
 
             for (int i = 0; i < count; i++) {
-                org.python.types.Bool b =
-                    (org.python.types.Bool) this.value.get(i).__lt__(otherList.value.get(i));
+                org.python.Object result = this.value.get(i).__lt__(otherList.value.get(i));
```

Comment:
@sjoerdjob Right, I mostly meant to direct question 1 at him.

I do agree, but this time I'll probably do it on `org.python.types.Object` as you point out, instead of the `org.python.Object` interface as I did on a previous iteration...